### ISSUE TITLE
Encountered warnings in Stage 1

### ISSUE THREAD

**Issue by Masterchiefm** [martinpacesa/BindCraft#122]
Hello, when i run the example in my local PC, I got these output in stage 1:

```
Stage 1: Test Logits
2024-12-06 14:31:56.084717: W external/xla/xla/service/gpu/ir_emitter_unnested.cc:1171] Unable to parse backend config for custom call: Could not convert JSON string to proto: Expected : between key:value pair.
= true, full_matrice
^
Fall back to parse the raw backend config str.
```

the command I used was `python -u ./bindcraft.py --settings './settings_target/PDL1.json' --filters './settings_filters/default_filters.json' --advanced './settings_advanced/default_4stage_multimer.json'`, which is copied from your repo, and the json files are the original files without modified. The PDL1.pdb is copied to the path described in json.

My OS is Ubuntu Desktop 24.04, with 32GB RAM and an Nvdia 3060.
Cuda version is 'Cuda compilation tools, release 12.0, V12.0.140'

However, the warning did not terminate the process. It seems calculating something but I don't know if the output is correct.

The full screen outputs:

```
(BindCraft) chief@chief-MS-7B89:~/BindCraft$ python -u ./bindcraft.py --settings './settings_target/PDL1.json' --filters './settings_filters/default_filters.json' --advanced './settings_advanced/default_4stage_multimer.json'
Available GPUs:
NVIDIA GeForce GTX 1660 SUPER1: gpu
┌──────────────────────────────────────────────────────────────────────────────┐
│                                 PyRosetta-4                                  │
│              Created in JHU by Sergey Lyskov and PyRosetta Team              │
│              (C) Copyright Rosetta Commons Member Institutions               │
│                                                                              │
│ NOTE: USE OF PyRosetta FOR COMMERCIAL PURPOSES REQUIRE PURCHASE OF A LICENSE │
│         See LICENSE.PyRosetta.md or email license@uw.edu for details         │
└──────────────────────────────────────────────────────────────────────────────┘
PyRosetta-4 2024 [Rosetta PyRosetta4.conda.ubuntu.cxx11thread.serialization.Ubuntu.python310.Release 2024.42+release.3366cf78a3df04339d1982e94531b77b098ddb99 2024-10-11T08:24:04] retrieved from: http://www.pyrosetta.org
Starting trajectory: PDL1_l65_s858892
Stage 1: Test Logits
2024-12-06 14:35:42.292398: W external/xla/xla/service/gpu/ir_emitter_unnested.cc:1171] Unable to parse backend config for custom call: Could not convert JSON string to proto: Expected : between key:value pair.
= true, full_matrice
^
Fall back to parse the raw backend config str.
2024-12-06 14:36:52.760675: W external/xla/xla/service/gpu/ir_emitter_unnested.cc:1171] Unable to parse backend config for custom call: Could not convert JSON string to proto: Expected : between key:value pair.
= true, full_matrice
^
Fall back to parse the raw backend config str.
1 models [3] recycles 1 hard 0 soft 0.02 temp 1 loss 11.75 helix 1.59 pae 0.78 i_pae 0.77 con 4.62 i_con 4.09 plddt 0.28 ptm 0.60 i_ptm 0.10 rg 10.05
2 models [1] recycles 1 hard 0 soft 0.04 temp 1 loss 9.53 helix 0.72 pae 0.70 i_pae 0.70 con 4.61 i_con 4.12 plddt 0.45 ptm 0.59 i_ptm 0.11 rg 1.89
3 models [4] recycles 1 hard 0 soft 0.05 temp 1 loss 13.61 helix 0.44 pae 0.69 i_pae 0.77 con 5.16 i_con 4.16 plddt 0.53 ptm 0.59 i_ptm 0.09 rg 13.23
4 models [3] recycles 1 hard 0 soft 0.07 temp 1 loss 8.62 helix 1.09 pae 0.67 i_pae 0.62 con 4.22 i_con 3.95 plddt 0.35 ptm 0.61 i_ptm 0.17 rg 1.16
5 models [4] recycles 1 hard 0 soft 0.09 temp 1 loss 8.18 helix 0.78 pae 0.62 i_pae 0.55 con 3.93 i_con 3.82 plddt 0.38 ptm 0.62 i_ptm 0.18 rg 0.87
6 models [4] recycles 1 hard 0 soft 0.11 temp 1 loss 7.66 helix 0.82 pae 0.62 i_pae 0.59 con 3.78 i_con 3.69 plddt 0.40 ptm 0.62 i_ptm 0.18 rg 0.11
7 models [3] recycles 1 hard 0 soft 0.13 temp 1 loss 7.58 helix 0.87 pae 0.60 i_pae 0.56 con 3.71 i_con 3.70 plddt 0.40 ptm 0.63 i_ptm 0.19 rg 0.11
8 models [1] recycles 1 hard 0 soft 0.14 temp 1 loss 7.30 helix 0.87 pae 0.57 i_pae 0.56 con 3.61 i_con 3.78 plddt 0.48 ptm 0.63 i_ptm 0.21 rg -0.66
9 models [2] recycles 1 hard 0 soft 0.16 temp 1 loss 6.94 helix 0.93 pae 0.54 i_pae 0.53 con 3.45 i_con 3.58 plddt 0.49 ptm 0.63 i_ptm 0.22 rg -0.56
10 models [3] recycles 1 hard 0 soft 0.18 temp 1 loss 6.71 helix 0.96 pae 0.52 i_pae 0.51 con 3.24 i_con 3.59 plddt 0.53 ptm 0.64 i_ptm 0.24 rg -0.58

```

**Comment by martinpacesa** [comment]
can you check the CUDA, flax, and jax version in your conda environment?

**Comment by Masterchiefm** [comment]
> can you check the CUDA, flax, and jax version in your conda environment?

Cuda: 12.2
Jax:0.4.36
flax:0.9.0

The warning didnot terminate the program, and it will show up in the beginning of each stage1.

**Comment by martinpacesa** [comment]
Can you try downgrading jax and jaxlib to 0.4.27 or below?

**Comment by HS-11** [comment]
I had the same issue, with 0.4.34 and below the error went away for me!

**Comment by Masterchiefm** [comment]
> Can you try downgrading jax and jaxlib to 0.4.27 or below?

 I tried to downgrade jax and jaxlib, then the problem changed to jax cannot recognize GPU. It seems the cuda should be downgraded, too. 

However, if I ignore the warning, the program still produced binders. 

It's annoying for config the cuda, so I'm going to ignore the warnings.

By the way, using the install_bindcraft.sh script will always install the latest cuda and   an incorrect jax, wich makes jax cannot recognize GPU.  I tried in 2 different PC and got the same jax problem. 

After review your script, I found that the script install the cpu jax, not the gpu jax. [The manual](https://jax.readthedocs.io/en/latest/installation.html) says that I need to know the cuda version(cuda12 for me) and then pip install **jax[cuda12]**  , instead of install jax. Thats why I will install the 0.4.35 jax.



Thanks a lot!
